dhis-2/dhis-services/dhis-service-administration/src/main/java/org/hisp/dhis/resourcetable/table/AbstractNameUniquenessAwareResourceTable.java

```diff
@@ -0,0 +1,60 @@
+package org.hisp.dhis.resourcetable.table;
+
+import static org.hisp.dhis.system.util.SqlUtils.quote;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.hisp.dhis.common.BaseDimensionalObject;
+import org.hisp.dhis.resourcetable.ResourceTable;
+
+/**
+ * This class can be extended by Resource Table generation classes that need to
+ * generate unique colum names
+ *
+ */
+public abstract class AbstractNameUniquenessAwareResourceTable<T>
+    extends
```

Comment:
I am wondering if it will be better to use a stateful utility class like `UniqueNameVerifier` or similar. Using the class hierarchy/subclassing for very specific stuff might be limiting/causing issues going forward.

Reply:
Yes, I initially created a small static method, but then I had to add the accumulator in each Resource class that needs uniqueness. Basically the could would look like:

```
int count = 0;
for ( DataElementGroupSet groupSet : objects )
{
    statement += quote( ensureUnique(groupSet.getShortName(), count) ) + " varchar(230), ";
    statement += quote( groupSet.getUid() ) + " character(11), ";
    count++;
}
```

With a super-class the accumulator is out of the way and the code is cleaner IMO. Report tables that do not need unique column names check, do not need to implement this class.

Reply:
Right, I meant we could something like:

```
UniqueValueVerifier verifier = new UniqueValueVerifier();
..
verifier.unique( groupSet.getShortName() );
```

Then it is just one more line to add (and we can remove the extends part), and we avoid using the class hierarchy.

But this is not a big deal, we can leave it as it is.

for ( DataElementGroupSet groupSet : objects )
{
  verifier.

Reply:
We could also use such a util class as part of an upgrade routine to add persisted short names to these entities.

Reply:
Ok, moved the logic to class.